### PR TITLE
[FIX] : Resolves ESLint Failures 🐛 

### LINKED ISSUE #169: Eslint is failing for pre commit hooks
Eslint is failing for pre commit hook and `lint-fix` is not working too

### PR DESCRIPTION
**Pull request by AdiAkhileshSingh15**
# Changes Overview
This PR aims to address unwanted ESlint failures and false positive errors by implementing the following changes:

- Adjusting the ESlint configuration.
- Downgrading the `eslint-formatter-pretty` package from v6 to v5. This adjustment is necessary because v6 is currently incompatible with ESlint due to its usage of cjs instead of esm. You can find more details about this issue [here](https://github.com/eslint/eslint/issues/15560).
- Making the necessary changes in the files to align with the ESlint rules.
- Executing `yarn run lint-fix` to identify and address any remaining ESlint errors by clearing out syntactical error warnings.
- Refactored existing test implementations to conform to ESLint-based code adjustments.

## Related Issue 🐛 
Fixes #169  

## Checklist ✅ 
Following each commit, the following checks were executed locally to ensure the functionality remains intact:

Commands used:
- [x] `yarn test` - All tests passed successfully.
- [x] `yarn dev` - No UI failures were observed.
- [x] `yarn build` - The build process completed without errors.
- [x] `yarn dev` - No new console errors were detected for any dependencies.

Additional checks:
- [x] Verified that the unit tests' functionality remained unchanged.
- [x] Reviewed all modifications made to files post `yarn run lint-fix`.
- [x] Consulted with the project maintainer regarding the introduced changes and obtained approval to proceed with the pull request.
- [x] Thoroughly tested all modifications locally.

## Reviewer
@lrasmus

### CODE REVIEW

**Comment by AdiAkhileshSingh15** [comment]
@lrasmus, I wanted to highlight the 3rd commit in this PR ([acf3c3d](https://github.com/StatTag/StatWrap/pull/197/commits/acf3c3dbf70f9cc1c4cfa280b0e3b37ff52bcbd8)), which focuses on resolving eslint issues using `lint-fix`. Since it involves numerous files and only addresses linting errors without introducing any breaking changes, I've already reviewed and confirmed each syntax correction. Hence, I recommend skipping a detailed review of this commit as it primarily consists of eslint syntax fixes.

File: app/utils/workflow.js [hunk 1]
```diff
@@ -46,9 +46,9 @@ export default class WorkflowUtil {
     // we don't count the 3 periods towards the max length (meaning, our resulting label will
     // technically go over the max length we detect).
     // eslint-disable-next-line prettier/prettier
-    const beginningPart = name.substring(0, (Constants.MAX_GRAPH_LABEL_LENGTH / 2));
+    const beginningPart = name.substring(0, Constants.MAX_GRAPH_LABEL_LENGTH / 2);
```
Comment (lrasmus) [inline app/utils/workflow.js:49]: I'd prefer not to use the linted code - this has an eslint-disable command that was added to explicitly avoid the linter suggestion.  This is a personal preference, but I feel strongly for this line of code the explicit parentheses should be included.

File: app/utils/workflow.js [hunk 2]
```diff
@@ -46,9 +46,9 @@ export default class WorkflowUtil {
...
     // eslint-disable-next-line prettier/prettier
-    const endPart = name.substring(name.length - (Constants.MAX_GRAPH_LABEL_LENGTH / 2) + 1);
+    const endPart = name.substring(name.length - Constants.MAX_GRAPH_LABEL_LENGTH / 2 + 1);
```
Comment (lrasmus) [inline app/utils/workflow.js:51]: I'd prefer not to use the linted code - this has an eslint-disable command that was added to explicitly avoid the linter suggestion.  This is a personal preference, but I feel strongly for this line of code the explicit parentheses should be included.

File: test/services/user.spec.js [hunk 3]
```diff
@@ -303,20 +303,20 @@ describe('services', () => {
             // to have something different.  Our next use of Date.now when
             // we add a new entry will then be the most recent time.
             // eslint-disable-next-line prettier/prettier
-            added: new Date(Date.now() - (10 * index)).toISOString()
+            added: new Date(Date.now() - 10 * index).toISOString(),
```
Comment (lrasmus) [inline test/services/user.spec.js:306]: I'd prefer not to use the linted code - this has an eslint-disable command that was added to explicitly avoid the linter suggestion.  This is a personal preference, but I feel strongly for this line of code the explicit parentheses should be included.

File: test/services/user.spec.js [hunk 4]
```diff
@@ -334,20 +334,20 @@ describe('services', () => {
...
```
Comment (lrasmus) [inline test/services/user.spec.js:337]: I'd prefer not to use the linted code - this has an eslint-disable command that was added to explicitly avoid the linter suggestion.  This is a personal preference, but I feel strongly for this line of code the explicit parentheses should be included.

File: test/utils/general.spec.js [hunk 5]
```diff
@@ -169,30 +169,30 @@ describe('services', () => {
       it('processes an array where the index field is a single value', () => {
         const inputArray = [
           { category: '1', value: 'a' },
-          { category: '1', value: 'b' }
+          { category: '1', value: 'b' },
         ];
-        expect(GeneralUtil.indexByField(inputArray, 'category')).toStrictEqual({ '1': inputArray });
+        expect(GeneralUtil.indexByField(inputArray, 'category')).toStrictEqual({ 1: inputArray });
```
Comment (lrasmus) [inline test/utils/general.spec.js:174]: It should preserve the original type here, the function does not convert the type from string to int.

File: test/utils/general.spec.js [hunk 6]
```diff
@@ -169,30 +169,30 @@ describe('services', () => {
...
       });
       it('processes an array where the index field can contain multiple values (array)', () => {
         const inputArray = [
           { category: ['1'], value: 'a' },
-          { category: ['1', '2'], value: 'b' }
+          { category: ['1', '2'], value: 'b' },
         ];
         expect(GeneralUtil.indexByField(inputArray, 'category')).toStrictEqual({
-          '1': [
+          1: [
```
Comment (lrasmus) [inline test/utils/general.spec.js:191]: It should preserve the original type here, the function does not convert the type from string to int.

File: test/utils/general.spec.js [hunk 7]
```diff
@@ -169,30 +169,30 @@ describe('services', () => {
...
             { category: ['1'], value: 'a' },
-            { category: ['1', '2'], value: 'b' }
+            { category: ['1', '2'], value: 'b' },
           ],
-          '2': [{ category: ['1', '2'], value: 'b' }]
+          2: [{ category: ['1', '2'], value: 'b' }],
```
Comment (lrasmus) [inline test/utils/general.spec.js:195]: It should preserve the original type here, the function does not convert the type from string to int.

File: app/services/projectList.js [hunk 8]
```diff
@@ -176,9 +176,9 @@ export default class ProjectListService {
     // is null and the other is a blank string, it's okay if they match.
     return projects.sort((a, b) => {
       // eslint-disable-next-line prettier/prettier
-      const stringA = (a && a.name) ? a.name.toLowerCase() : Constants.UndefinedDefaults.PROJECT;
+      const stringA = a && a.name ? a.name.toLowerCase() : Constants.UndefinedDefaults.PROJECT;
```
Comment (lrasmus) [inline app/services/projectList.js:179]: I'm okay to accept the linter suggestion here, but we should remove the eslint-disable line on 178

File: app/services/projectList.js [hunk 9]
```diff
@@ -176,9 +176,9 @@ export default class ProjectListService {
...
       // eslint-disable-next-line prettier/prettier
-      const stringB = (b && b.name) ? b.name.toLowerCase() : Constants.UndefinedDefaults.PROJECT;
+      const stringB = b && b.name ? b.name.toLowerCase() : Constants.UndefinedDefaults.PROJECT;
```
Comment (lrasmus) [inline app/services/projectList.js:181]: I'm okay to accept the linter suggestion here, but we should remove the eslint-disable line on 180

File: app/services/sourceControl.js [hunk 10]
```diff
@@ -38,7 +38,7 @@ export default class SourceControlService {
     }
 
     // eslint-disable-next-line prettier/prettier
-    return new Date((committer.timestamp - (committer.timezoneOffset * 60)) * 1000);
+    return new Date((committer.timestamp - committer.timezoneOffset * 60) * 1000);
```
Comment (lrasmus) [inline app/services/sourceControl.js:41]: I'd prefer not to use the linted code - this has an eslint-disable command that was added to explicitly avoid the linter suggestion.  This is a personal preference, but I feel strongly for this line of code the explicit parentheses should be included.

File: app/utils/project.js [hunk 11]
```diff
@@ -172,7 +172,11 @@ export default class ProjectUtil {
           f.category === Constants.FilterCategory.FILE_TYPE
         ) {
           // eslint-disable-next-line prettier/prettier
-          const catFilter = f.values.some(v => !v.value && WorkflowUtil.getAssetType(asset) === v.key) ? 2 : 0;
+          const catFilter = f.values.some(
```
Comment (lrasmus) [inline app/utils/project.js:175]: I'm okay to accept the linter suggestion here, but we should remove the eslint-disable line on 174

File: app/utils/workflow.js [hunk 12]
```diff
@@ -210,8 +210,12 @@ export default class WorkflowUtil {
       };
       // Only push dependencies once (to avoid unnecessary clutter)
       // eslint-disable-next-line prettier/prettier
-      if (!tree.children.some(x => x.name === depEntry.name
-            && x.attributes.assetType === depEntry.attributes.assetType)) {
+      if (
```
Comment (lrasmus) [inline app/utils/workflow.js:213]: I'm okay to accept the linter suggestion here, but we should remove the eslint-disable line on 212

File: test/services/assets/handlers/python.spec.js [hunk 13]
```diff
@@ -427,65 +427,67 @@ describe('services', () => {
...
-          new PythonHandler().getOutputs('test.uri', '# f = open("test.dat", "w")').length
+          new PythonHandler().getOutputs('test.uri', '# f = open("test.dat", "w")').length,
         ).toEqual(0);
         expect(
-          new PythonHandler().getOutputs('test.uri', "   #df.to_csv('test.csv')").length
+          new PythonHandler().getOutputs('test.uri', "   #df.to_csv('test.csv')").length,
         ).toEqual(0);
         // eslint-disable-next-line prettier/prettier
-        expect(new PythonHandler().getOutputs('test.uri', '   #    f = open("test.dat", "w")').length).toEqual(0);
+        expect(
```
Comment (lrasmus) [inline test/services/assets/handlers/python.spec.js:488]: I'm okay to accept the linter suggestion here, but we should remove the eslint-disable line on 487

File: test/services/assets/handlers/sas.spec.js [hunk 14]
```diff
@@ -486,65 +486,67 @@ describe('services', () => {
...
-          new SASHandler().getOutputs('test.uri', '* ods csvall file="test.pdf";').length
+          new SASHandler().getOutputs('test.uri', '* ods csvall file="test.pdf";').length,
         ).toEqual(0);
         expect(
-          new SASHandler().getOutputs('test.uri', ' *ods excel file="test.xlsx";').length
+          new SASHandler().getOutputs('test.uri', ' *ods excel file="test.xlsx";').length,
         ).toEqual(0);
         // eslint-disable-next-line prettier/prettier
-        expect(new SASHandler().getOutputs('test.uri', '  *   ods excel file="test.xlsx";  ').length).toEqual(0);
+        expect(
```
Comment (lrasmus) [inline test/services/assets/handlers/sas.spec.js:542]: I'm okay to accept the linter suggestion here, but we should remove the eslint-disable line on 541

File: test/services/projectList.spec.js [hunk 15]
```diff
@@ -114,14 +114,20 @@ describe('services', () => {
         const service = new ProjectListService();
         expect(() => service.validateProjectListEntry({ path: '/Test/Path' })).toThrow(Error);
         // eslint-disable-next-line prettier/prettier
-        expect(() => service.validateProjectListEntry({ id: null, path: '/Test/Path' })).toThrow(Error);
+        expect(() => service.validateProjectListEntry({ id: null, path: '/Test/Path' })).toThrow(
```
Comment (lrasmus) [inline test/services/projectList.spec.js:117]: I'm okay to accept the linter suggestion here, but we should remove the eslint-disable line above

File: test/services/projectList.spec.js [hunk 16]
```diff
@@ -114,14 +114,20 @@ describe('services', () => {
...
+          Error,
+        );
         expect(() =>
-          service.validateProjectListEntry({ id: undefined, path: '/Test/Path' })
+          service.validateProjectListEntry({ id: undefined, path: '/Test/Path' }),
         ).toThrow(Error);
         // eslint-disable-next-line prettier/prettier
-        expect(() => service.validateProjectListEntry({ id: '', path: '/Test/Path' })).toThrow(Error);
+        expect(() => service.validateProjectListEntry({ id: '', path: '/Test/Path' })).toThrow(
```
Comment (lrasmus) [inline test/services/projectList.spec.js:124]: I'm okay to accept the linter suggestion here, but we should remove the eslint-disable line above

File: test/services/projectList.spec.js [hunk 17]
```diff
@@ -114,14 +114,20 @@ describe('services', () => {
...
+          Error,
+        );
         // eslint-disable-next-line prettier/prettier
-        expect(() => service.validateProjectListEntry({ id: '   ', path: '/Test/Path' })).toThrow(Error);
+        expect(() => service.validateProjectListEntry({ id: '   ', path: '/Test/Path' })).toThrow(
```
Comment (lrasmus) [inline test/services/projectList.spec.js:128]: I'm okay to accept the linter suggestion here, but we should remove the eslint-disable line above

File: test/services/projectList.spec.js [hunk 18]
```diff
@@ -136,7 +142,9 @@ describe('services', () => {
       it('should not throw an error if the ID and path are specified', () => {
         const service = new ProjectListService();
         // eslint-disable-next-line prettier/prettier
-        expect(() => service.validateProjectListEntry({ id: '1', path: '/Test/Path' })).not.toThrow(Error);
+        expect(() => service.validateProjectListEntry({ id: '1', path: '/Test/Path' })).not.toThrow(
```
Comment (lrasmus) [inline test/services/projectList.spec.js:145]: I'm okay to accept the linter suggestion here, but we should remove the eslint-disable line above

File: test/services/sourceControl.spec.js [hunk 19]
```diff
@@ -76,42 +76,43 @@ describe('services', () => {
...
-        expect(service.getFileHistory('/test/path', null)).resolves.toBeNull();
+        await expect(service.getFileHistory('/test/path', null)).resolves.toBeNull();
         return expect(service.getFileHistory('/test/path', undefined)).resolves.toBeNull();
       });
 
       it('should return an empty array for invalid file path', async () => {
         const service = new SourceControlService();
-        // eslint-disable-next-line prettier/prettier
-        return expect(service.getFileHistory('/Invalid/Path', 'DoesNotExist.txt')).resolves.toStrictEqual([]);
+        return expect(
```
Comment (lrasmus) [inline test/services/sourceControl.spec.js:106]: I'm okay to accept the linter suggestion here, but we should remove the eslint-disable line above

File: app/main.dev.js [hunk 20]
```diff
@@ -670,11 +671,22 @@ ipcMain.on(Messages.LOAD_PROJECT_LOG_REQUEST, async (event, project) => {
 ipcMain.on(
   Messages.UPDATE_PROJECT_REQUEST,
   // eslint-disable-next-line prettier/prettier
-  async (event, projectPath, actionType, entityType, entityKey, title, description, details, level, user) => {
+  async (
```
Comment (lrasmus) [inline app/main.dev.js:674]: I'm okay to accept the linter suggestion here, but we should remove the eslint-disable line above

**Review by lrasmus — CHANGES_REQUESTED** [review]
Overall I'm good with the linter changes, but there are some places I disagree with the linter (and I had an explicit override in place that it missed), and some other places I'm okay to accept the linter's suggestion, but we just need to remove an override comment that was in place.

File: test/utils/general.spec.js (hunk 7 above)
Comment (AdiAkhileshSingh15) [reply test/utils/general.spec.js:195]: I guess integers should work fine here. Both integers and strings are acceptable for JSON fields.
I've added an example codesnip here.
Screenshot from 2024-04-03 13-09-17

File: app/utils/workflow.js (hunk 1 above)
Comment (AdiAkhileshSingh15) [reply app/utils/workflow.js:49]: I understand you're not suggesting that it alters the functionality, but rather, you prefer to keep the parentheses for the sake of code clarity. Am I correct though?

File: test/utils/general.spec.js (hunk 7 above)
Comment (AdiAkhileshSingh15) [reply test/utils/general.spec.js:195]: Oh understood!

Comment (AdiAkhileshSingh15) [reply test/utils/general.spec.js:195]: But, I guess integers should work fine here. Both integers and strings are acceptable for JSON fields. And, the function is returning a JSON object.
Added an example codesnip here.
Screenshot from 2024-04-03 13-09-17
Please correct me if am wrong though.

**Comment by AdiAkhileshSingh15** [comment]
@lrasmus do check the introduction of `title` dependency in `app/components/EditableSelect/EditableSelect.js` useEffect.

File: app/utils/workflow.js (hunk 1 above)
Comment (lrasmus) [reply app/utils/workflow.js]: Correct!  The code continues to work fine, it's just my preference on style for code clarity :)  Thanks!

**Comment by lrasmus** [comment]
> @lrasmus do check the introduction of `title` dependency in `app/components/EditableSelect/EditableSelect.js` useEffect.

Thanks for mentioning that!  I did notice it, and when testing it didn't seem to impact the functionality (from what I could tell).  Curious if you know why the linter recommended it, and how strong the recommendation is?  I'm okay keeping it if there is a good reason.  I'm equally okay ignoring the linter's suggestion there and reverting to how it was before.

**Comment by AdiAkhileshSingh15** [comment]
@lrasmus I've updated the branch with latest changes in master.
It's ready to merge, if all reviews are done.

File: test/utils/general.spec.js (hunk 7 above)
Comment (lrasmus) [reply test/utils/general.spec.js:195]: You are correct that integers and strings are okay.  However, the function is taking the category values ('1' as a string) and returning that as part of the response.  So the function is not changing '1' to 1, and the test should reflect that we expect '1' returned as a string.

**Comment by AdiAkhileshSingh15** [comment]
@lrasmus please check latest push

**Comment by lrasmus** [comment]
Amazing, thank you so much @AdiAkhileshSingh15 !

**Review by lrasmus — APPROVED** [review]

**Pull request merged**